### DISCUSSION TITLE
Do particles get destroyed when they run out of life?

### DISCUSSION THREAD

**Discussion by longnguyen2004** [matteobruni/tsparticles#5100] [Q&A]
I'm trying to make an effect where particles get emitted and stay on screen for 2-3 seconds before going away. My `life` config is setup like so

```js
{
    // other stuff
    life: {
        count: 1,
        duration: {
            value: {
                min: 2,
                max: 3
            }
        }
    }
}
```

The effect works fine, but I noticed that memory usage was going up non-stop, which makes me believe that the particle isn't destroyed when its life runs out. Is this a bug or intended behavior, and if so, what's the correct way to implement that effect?

(And just to be sure I didn't mess up anything, I removed the `life` config and just let the particles go off screen, and they do get destroyed normally)

**Comment by matteobruni** [comment]
Yes particles are destroyed, I've created this CodePen where the particles count can be tracked: https://codepen.io/matteobruni/pen/wvQNYJz?editors=0010

I've noticed using this sample that the pool of destroyed particles is not updated when using life options. Thanks for that! I've tested that CodePen on Chrome checking the memory snapshot but every time I checked it, it was always the same amount of memory used. Maybe there's something in other part of your code that is affecting memory?

**Reply by longnguyen2004** [reply]
It was the `_zArray` array that eventually leak. I modified your pen to produce 100 particles every 0.1 second and remove the emitter `life`. It didn't take long for me to reproduce the issue.

![image](https://github.com/matteobruni/tsparticles/assets/15729831/40ffa412-011a-4864-9b13-9e56fe214bb7)

(Subframe is the frame that has tsParticles)

**Reply by matteobruni** [reply]
You're right, I've added the tracking to that array too and it's not managed well. Thanks for having a look

**Reply by longnguyen2004** [reply]
I'll make an issue out of this for tracking purposes.

**Comment by longnguyen2004** [comment]
Will be fixed in https://github.com/matteobruni/tsparticles/pull/5071
